### PR TITLE
Simplified test for yast_i handling of the norecommends-option

### PR DESCRIPTION
**Pull request by cwh42**
As a followup to #646 with these improvements:
- respect new value PACKAGETOINSTALLWITHRECOMMENDS if present
- defaults to package powertop
- simplified (because of better package name)
- based on old test to reduce diff size

### CODE REVIEW

**Comment by cwh42** [comment]
Find needed needles in https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/43

**Comment by DimStar77** [comment]
:+1:

**Comment by sysrich** [comment]
@cwh42 where can I find the sle12 needles for this test? I dont see a PR on gitlab.suse.de

**Comment by nilxam** [comment]
IIRC the PR was merged already on gitlab

File: tests/console/yast2_i.pm [hunk 1]
```diff
@@ -3,7 +3,7 @@ use testapi;
 
 sub run() {
     my $self    = shift;
-    my $pkgname = get_var("PACKAGETOINSTALL");
+    my $pkgname = get_var("PACKAGETOINSTALLWITHRECOMMENDS") || 'powertop';
```
Comment (lnussel) [inline tests/console/yast2_i.pm]: get_var("PACKAGETOINSTALLWITHRECOMMENDS",  'powertop');

**Comment by cwh42** [comment]
@sysrich @nilxam The needles for the new version trying to install 'powertop' are still mission for sle12. Now that I tried it for sle I noticed, that there is no powertop for sle :(

**Comment by nilxam** [comment]
we can set PACKAGETOINSTALLWITHRECOMMENDS=mc to openqa.s.d then, @sysrich what do you think?

File: tests/console/yast2_i.pm (hunk 1 above)
Comment (cwh42) [inline tests/console/yast2_i.pm]: Fixed

**Comment by cwh42** [comment]
I found 'dasher' as an appropriate package for SLE.
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/20
main.pm for SLE still needs to be changed:
set_var("PACKAGETOINSTALLWITHRECOMMENDS", "dasher");
Sorry, I cannot - my sle branch checkout is completely messed up.

**Comment by ancorgs** [comment]
I'm taking over @cwh42 openQA tasks while he is on vacation.

What's the current status of this PR? I tested with Build20150811 and it works, but I found the PR with the needles to include a lot of unrelated stuff. Thus, I created this alternative PR which includes only needles relevant to this PR.
https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/44

I'll take a look to the SLE needles now (they seem to have the same problem).

**Comment by cwh42** [comment]
@ancorgs there should obly be two needles for SLE in this commit https://gitlab.suse.de/cwh/os-autoinst-needles-sles/commit/1742d40169e170d9d1bbc5101997554e99b83265

**Comment by ancorgs** [comment]
The change needed to have this running in SLE12 is here
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/677
The corresponding needles are here
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/21
The needles for openSUSE are already merged

Last but not least, an example of successful execution in SLES12
http://yast-openqa.suse.cz/tests/1

Is there something else preventing this to be merged?

**Comment by coolo** [comment]
Richard's vacation

**Comment by ancorgs** [comment]
I have just found that powertop doesn't fit the requisites in openSUSE:42. It doesn't seem to be a powertop-lang package there. Any suggestion?

**Comment by ancorgs** [comment]
Actually, dasher seems to be present in TW, 42 and sle12. I'll run some experiments to check if we can simply use dasher everywhere (although I guess @cwh42 already tried before introducing the difference in SLE).

**Comment by coolo** [comment]
it also has to be in the staging projects, i.e. openSUSE:42:Rings:1-MinimalX

**Comment by ancorgs** [comment]
Ok. So empirically checked that dasher is unsurprisingly not in 42:rings:1 https://yast-openqa.suse.cz/tests/24

@coolo any suggestion? Nobody knows the rings better than you

**Comment by mvidner** [comment]
yast2-kdump recommends kdump and makedumpfile, and all are part of https://build.opensuse.org/project/show/openSUSE:42:Rings:1-MinimalX . Would that work?

**Comment by ancorgs** [comment]
@mvidner. Hopefully. I will confirm it in a couple of hours :-P

**Comment by mvidner** [comment]
This time for sure™ - yast2-buildtools (subpackage of yast2-devtools) recommends cmake, doxygen, and libtool.

**Comment by ancorgs** [comment]
Actually, with the current implementation of the test there would be even an additional requirement:

* The name of the suggested package includes the name of the searched package

Time to improve the test, I guess.

**Comment by ancorgs** [comment]
@mvidner, @coolo Before going for my vacation, here you are an alternative implementation that does not rely in the packages being called almost the same. It should be easier to find a pair of packages that satisfy the requisites in all scenarios
https://github.com/ancorgs/os-autoinst-distri-opensuse/commit/2011067613ae7a7f9d904cd32685a38545d077dd

See it in action here.
https://yast-openqa.suse.cz/tests/68
As you can see, it almost works but it also brings several other additional packages in. It's up to you if you want to look for another pair of `$pkgname` `$recommended` with fewer dependencies or to accept the additional packages.

**Comment by ancorgs** [comment]
Bad news, no `yast2-buildtools` in openSUSE:42:Rings:1-MinimalX
https://yast-openqa.suse.cz/tests/73/

**Comment by mvidner** [comment]
I have succeeded with yast2-nfs-client -> nfs-client in http://yast-openqa.suse.cz/tests/98, will make PRs for that.

**Comment by mvidner** [comment]
This PR is superseded by #686 and can be closed.

**Pull request closed without merging**